tensorflow/contrib/periodic_intersperse/core/kernels/periodic_intersperse_op.h

```diff
@@ -0,0 +1,216 @@
+#ifndef TENSORFLOW_KERNELS_PERIODICINTERSPERSE_OP_H_
+#define TENSORFLOW_KERNELS_PERIODICINTERSPERSE_OP_H_
+
+#include <cmath>
+#include <type_traits>
+#include <vector>
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/shape_inference.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+
+using namespace tensorflow;
+using std::vector;
+using std::decay;
+
+
+template<class IndexVec_t, class Index_t>
```

Comment:
Template type names should be CamelCase: https://google.github.io/styleguide/cppguide.html#Type_Names